pkg/parallel/parallel_linux.go

```diff
@@ -0,0 +1,57 @@
+package parallel
+
+import (
+	"context"
+	"sync"
+
+	"github.com/containers/libpod/libpod"
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+)
+
+// ParallelContainerOp performs the given function on the given set of
+// containers, using a number of parallel threads.
+// If no error is returned, each container specified in ctrs will have an entry
+// in the resulting map; containers with no error will be set to nil.
+func ParallelContainerOp(ctx context.Context, ctrs []*libpod.Container, applyFunc func(*libpod.Container) error) (map[*libpod.Container]error, error) {
+	jobControlLock.RLock()
+	defer jobControlLock.RUnlock()
+
+	// We could use a sync.Map but given Go's lack of generic I'd rather
+	// just use a lock on a normal map...
+	// The expectation is that most of the time is spent in applyFunc
+	// anyways.
+	var (
+		errMap  map[*libpod.Container]error = make(map[*libpod.Container]error)
+		errLock sync.Mutex
+		allDone sync.WaitGroup
+	)
+
+	for _, ctr := range ctrs {
```

Comment:
Potential future optimization: partition the `ctrs` slice and let goroutines operate on sub-slices (partitions) to reduce the overhead of creating goroutines.